### PR TITLE
Adding cpu and memory configuration

### PR DESCRIPTION
**Pull request by felixp72**
Signed-off-by: Felix Puscasu <fpuscasu@amazon.com>

### CODE REVIEW

File: src/fc/fc_monitor.c [hunk 1]
```diff
@@ -154,6 +155,41 @@ virFCJsonActionExec(const char *unix_path,
...
+    int mem_mb = virDomainDefGetMemoryInitial(vmdef);
+    int vcpu_count = 0;
+    int maxvcpus = 0;
+    virDomainVcpuDef *vcpu;
+    size_t i;
+
+    VIR_DEBUG("Memory in mb: %d", mem_mb);
+
+    /* count maximum allowed number vcpus and enabled vcpus when boot.*/
+    maxvcpus = virDomainDefGetVcpusMax(vmdef);
```
Comment (alindima) [inline src/fc/fc_monitor.c:174]: Firecracker does not support scaling the number of vcpus up or down. Can we enforce this while validating the XML?

File: src/fc/fc_driver.c [hunk 2]
```diff
@@ -369,6 +369,8 @@ fcDomainGetInfo(virDomainPtr domain,
     info->maxMem = virDomainDefGetMemoryTotal(vm->def);
     info->memory = info->maxMem;
 
+    info->state = virDomainObjGetState(vm, NULL);
```
Comment (alindima) [inline src/fc/fc_driver.c]: why is this needed? I see it's already called in this function

File: src/fc/fc_monitor.c (hunk 1 above)
Comment (felixp72) [reply src/fc/fc_monitor.c:174]: in the same fashion as for memory, the `<vcpu> [number] </vcpu>` tag specifies the maximum possible vcpus that can be used. Specifying that currently there are less vcpus is done with an attribute of `current=[number]`  set to the `<vcpu>` tag. We can skip the for loop below and just use the maximum vcpus given in the definition. 
This should also be mentioned in the docs

File: src/fc/fc_monitor.c [hunk 3]
```diff
@@ -154,6 +155,34 @@ virFCJsonActionExec(const char *unix_path,
...
 }
 
+int virFCMonitorSetConfig(const char *socketpath,
+                          bool hyper_threading,
+                          virDomainDef *vmdef)
+{
+    virJSONValue_autoptr jsonObj = virJSONValueNewObject();
+    long response_code = 0;
+    int ret = -1;
+    int mem_mb = virDomainDefGetMemoryInitial(vmdef);
```
Comment (alindima) [inline src/fc/fc_monitor.c]: what is the unit returned by this? isn't it in bytes?

File: src/fc/fc_domain.c [hunk 4]
```diff
@@ -386,7 +390,7 @@ fcConfigAndStartVM(virFCDriver *driver G_GNUC_UNUSED,
     }
 
     if (virFCMonitorSetDisk(vmPrivateData->socketpath, "rootfs",
-                            root_device->src->path, true, false) < 0) {
+                            root_device->src->path, true, root_device->src->readonly) < 0) {
```
Comment (felixp72) [inline src/fc/fc_domain.c:429]: if you add the tag `<readonly/>` under the `<disk>` device definition, the rootfs will be in read-only mode

**Review by alindima — APPROVED** [review]

**Review by gbionescu — APPROVED** [review]
